src/libImaging/Jpeg2KDecode.c

```diff
@@ -742,8 +742,8 @@ j2k_decode_entry(Imaging im, ImagingCodecState state)
            swapped), bail. */
         if (tile_info.x0 >= tile_info.x1
             || tile_info.y0 >= tile_info.y1
-            || tile_info.x0 < image->x0
-            || tile_info.y0 < image->y0
+            || tile_info.x0 < (OPJ_INT32)image->x0
+            || tile_info.y0 < (OPJ_INT32)image->y0
             || tile_info.x1 - image->x0 > im->xsize
             || tile_info.y1 - image->y0 > im->ysize) {
```

Comment:
```suggestion
            || (OPJ_INT32)(tile_info.x1 - image->x0) > im->xsize
            || (OPJ_INT32)(tile_info.y1 - image->y0) > im->ysize) {
```

The previous 4 lines verify that `tile_info.x1 > tile_info.x0` and `tile_info.x0 >= image->x0`, so by transitivity `tile_info.x1 > image->x0` and so the result fits into the unsigned type of `tile_info.x1` (`y` is analogous).